### DISCUSSION TITLE
Submit notice return 500

### DISCUSSION THREAD

**Discussion by user6330** [OP-TED/eForms-SDK#364] [Q&A] [answered]
I'm using PREW 'https://enotices2.preview.ted.europa.eu/esenders/api/v2/notice/notices/submit' to test the submission of the xml already validated by PREW CVS.

I converted the **CURL** used in the SWAGGER UI

`curl -X 'POST' \
  'https://enotices2.preview.ted.europa.eu/esenders/api/v2/notice/notices/submit' \
  -H 'accept: application/json' \
  -H 'Authorization: Bearer xxxxxxxxxxxxxx' \
  -H 'Content-Type: multipart/form-data' \
  -F 'notice=@file.xml;type=text/xml' \
  -F 'metadata={
  "noticeAuthorEmail": "test@test.com",
  "noticeAuthorLocale": "en"
}'`

in **JAVA**

```
OkHttpClient client = new OkHttpClient();
        MediaType mediaType = MediaType.parse("text/xml");
        RequestBody requestBody = new MultipartBody.Builder()
                .setType(MultipartBody.FORM)
                .addFormDataPart("notice", "file.xml;",
                        RequestBody.create(new File("path/to/file.xml;"), mediaType))
                .addFormDataPart("metadata", "{\n  \"noticeAuthorEmail\": \"test@test.com\",\n  \"noticeAuthorLocale\": \"en\"\n}")
                .build();
        Request request = new Request.Builder()
                .url("https://enotices2.preview.ted.europa.eu/esenders/api/v2/notice/notices/submit")
                .header("accept", "application/json")
                .header("Authorization", "Bearer xxxxxxxxxxxxxxxxxxxxx")
                .post(requestBody)
                .build();
        Response response = client.newCall(request).execute();
        System.out.println(response.body().string());
```

The response is "**500 - Internal error**" instead of the same response in the SWAGGER UI.

**Comment by pcolast** [comment]
Hello,

If I understand correctly, the submit is successful when done in the SWAGGER UI, correct ?
Apart from "500 - Internal error" what is the message in the response body ?
Can you repeat the failing test and provide the timestamp ?
Can you also provide the notice used in the test ?

Thank you.
Have a nice evening.
Stefan

**Reply by rengit62** [reply]
Stefan,
today I do not get error 500 , instead I get 401 This API Key does not exist with this curl request ( made by swagger UI )
curl -X 'POST' \
  'https://enotices2.ted.europa.eu/esenders/api/v2/notice/notices/submit' \
  -H 'accept: application/json' \
  -H 'Authorization: Bearer xxx' \
  -H 'Content-Type: multipart/form-data' \
  -F 'notice=@noticeEncoded.txt;type=text/plain' \
  -F 'metadata={
  "noticeAuthorEmail": "rbaviera@jaggaer.com",
  "noticeAuthorLocale": "en"
}'

With PostMan with **same api-key**  posting a notice to validation I get 200 return code and the schematron file 
![image](https://user-images.githubusercontent.com/52211037/224250610-c268eea0-d209-4b10-8427-aaa305213919.png)

The api-key for submitting a notice is different from the api-key to validating ?
Thanks
Renato

**Reply by rouschr** [reply]
Note that your API key was public for a while, I edited your post and put xxx. I recommend you create a new key. The API key is still in the screenshot.

**Reply by user6330** [reply]
@pcolast here is the response 

`<!doctype html><html lang="en"><head><title>HTTP Status 500 – Internal Server Error</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 500 – Internal Server Error</h1></body></html>`


the notice i used via SWAGGER UI return this response.

![image](https://user-images.githubusercontent.com/127392893/224344335-5a80b02b-3a48-4b6d-85e6-170a97169157.png)


I'm expecting at least the same kind of results

**Reply by user6330** [reply]
Via RESTer (Chrome plugin) here is the request and the response

![image](https://user-images.githubusercontent.com/127392893/224350827-0094ab12-e214-4c5b-bbf2-9a7887d9cf88.png)

![image](https://user-images.githubusercontent.com/127392893/224351197-d8e8f5ed-c465-475a-b69b-c4574d128bfe.png)

**Comment by pcolast** [comment]
Hello,

So the API call with java code gives error 500, whereas the API call via swagger or RESTer (Chrome plugin) gives error 400 when using the same parameters.
Can you please execute the API call with java code again and provide the timestamp ?
This would help us to find the received request and make a comparison.

Thank you.

Stefan

**Reply by user6330** [reply]
Hello,

The timestamp get from the Response is **"1678720748664"**. 
In Date format is **"Mon Mar 13 16:19:08 CET 2023"**.

**Comment by pcolast** [comment, marked as answer]
Hello,

We found an error in the logs
2023-03-13T15:19:06.984Z [E] [13-03-2023 16:19:06.984 CET] [http-nio-8080-exec-41] [05e624ca-ea2b-463c-bc99-1f1d703fc54b] [eu.europa.ted.enotices2.notice.exception.RestExceptionHandler] [Unexpected error]
2023-03-13T15:19:06.984Z java.lang.RuntimeException: org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 1; Premature end of file.

Could you try to remove the semicolon
so instead of RequestBody.create(new File("path/to/file.xml;"), mediaType)
use RequestBody.create(new File("path/to/file.xml"), mediaType)

Have a nice day,
Stefan

**Reply by user6330** [reply]
Hello Stefan,

With the error logs I found out that the problem was in the File I pass to the Request body.
Now I get the json that I expect.


Thanks for your answer.
Ferdinand
